src/__tests__/identity_flow.test.ts

```diff
@@ -0,0 +1,187 @@
+    const role1 = await client.createRealmRole(realmName, {
+      name: 'FridgeAccess',
+      description: 'Grants access to the secrets of the fridge.',
+    })
+    const role2 = await client.createRealmRole(realmName, {
+      name: 'StovePowers',
+      description: 'They can turn on the stove.',
+    })
+
+    // Expected there are no mapped roles to start
+    const roleMappings = await client.listGroupRoleMappings(realmName, group.id)
+    expect(roleMappings.client).toEqual({})
+    expect(roleMappings.realm).toBeInstanceOf(Array)
+    expect(roleMappings.realm).toHaveLength(0)
+
+    // add a mapping to the group
+    const addResponse = await client.addGroupRoleMappings(realmName, group.id, {
+      realm: [
+        // use minimal set of inputs
+        { id: role1.id, name: role1.name, description: role1.description },
+        // or a full blown Role
+        role2,
+      ],
+    })
+
+    expect(addResponse).toBeTruthy()
+
+    // List out the groups the identity is in, expected only ToznyEngineers
+    const responseToznyEngineers = await client.groupMembership(
+      realmName,
+      idDetails.toznyId
+    )
+    // Check to make sure the group we added is the only group
+    expect(responseToznyEngineers).toBeInstanceOf(Array)
+    expect(responseToznyEngineers).toHaveLength(1)
+    expect(responseToznyEngineers[0].id).toBeTruthy()
+    expect(responseToznyEngineers[0].name).toBe('ToznyEngineers')
+
+    // Get Identity Details to check current roles
+    const identityDetails = await client.identityDetails(
+      realmName,
+      sovereignName.toLowerCase()
+    )
+    expect(identityDetails.id).toBeTruthy()
+    expect(identityDetails.toznyId).toBeTruthy()
+    expect(identityDetails).toBeInstanceOf(DetailedIdentity)
+    expect(identityDetails.roles.realm).toBeInstanceOf(Array)
+    // 2 roles we added from the group and 2 default roles
+    expect(identityDetails.roles.realm).toHaveLength(4)
+
+    // Leave the group
+    await client.leaveGroups(realmName, idDetails.toznyId, {
+      groups: [group.id],
+    })
+
+    // Check group membership expected to be 0
+    const epmty = await client.groupMembership(realmName, idDetails.toznyId)
+    expect(epmty).toBeInstanceOf(Array)
+    expect(epmty).toHaveLength(0)
+
+    // Get Identity Details to check current roles after group removal
+    const identityDetailsAfterGroup = await client.identityDetails(
+      realmName,
+      sovereignName.toLowerCase()
+    )
+    expect(identityDetailsAfterGroup.id).toBeTruthy()
+    expect(identityDetailsAfterGroup.toznyId).toBeTruthy()
+    expect(identityDetailsAfterGroup).toBeInstanceOf(DetailedIdentity)
+    expect(identityDetailsAfterGroup.roles.realm).toBeInstanceOf(Array)
+    //  2 default roles expected after group removal
+    expect(identityDetailsAfterGroup.roles.realm).toHaveLength(2)
+
+    // Delete new identity
+    await client.deleteIdentity(realmName, identityResponse.identity.tozny_id)
```

Comment:
use the fetched `identityDetails` rather than the registration event identity wherever you require the `toznyId`